classes/form/section_select_form.php

```diff
@@ -87,6 +88,24 @@ public function definition() {
             $radioarray[] = $mform->createElement('radio', 'targetsectionnum',
                 '', $sectionname, $sectionnum, ['class' => 'mt-2']);
         }
+
+        $course = $DB->get_record('course', array('id' => $targetcourseid), '*', MUST_EXIST);
+        $courseformat = course_get_format($course);
+        $courseformatoptions = $courseformat->get_format_options();
+        $maxsections = $courseformat->get_max_sections();
+        $desirednumsections = 0;
+
+        if (isset($courseformatoptions['numsections'])) { // Course format setting.
+            $desirednumsections = $courseformatoptions['numsections'] + 1;
+        } else if (course_get_format($course)->uses_sections()) { // No format setting.
+            $desirednumsections = $courseformat->get_last_section_number() + 1;
+        }
+
+        if ($desirednumsections <= $maxsections) { // New section option.
```

Comment:
Currently the ability to add a section to a course is bound to the capability `moodle/course:update`. We therefore should only show this option if the user has this capability. Maybe we should also doublecheck that capability in the function `duplicate_to_course` if a new section has been requested.

Reply:
yep, you are correct